pkg/cmd/workflow/view/view.go

```diff
@@ -0,0 +1,253 @@
+		  # Interactively select a workflow to view
+		  $ gh workflow view
+
+		  # View a specific workflow
+		  $ gh workflow view 0451
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+
+			opts.Raw = !opts.IO.CanPrompt()
+
+			if len(args) > 0 {
+				opts.WorkflowID = args[0]
+			} else if !opts.IO.CanPrompt() {
+				return &cmdutil.FlagError{Err: errors.New("workflow ID required when not running interactively")}
+			} else {
+				opts.Prompt = true
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return runView(opts)
+		},
+	}
+	cmd.Flags().BoolVarP(&opts.Web, "web", "w", false, "Open workflow in the browser")
+
+	return cmd
+}
+
+func runView(opts *ViewOptions) error {
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("could not build http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+
+	repo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("could not determine base repo: %w", err)
+	}
+
+	var workflow *shared.Workflow
+
+	if opts.Prompt {
+		workflow, err = promptWorkflows(client, repo)
+		if err != nil {
+			return err
+		}
+	}
+
+	if workflow == nil {
+		workflow, err = resolveWorkflow(opts.IO, client, repo, opts.WorkflowID)
+		if err != nil {
+			return err
+		}
+	}
+
+	if opts.Web {
+		hostname := repo.RepoHost()
+		baseName := filepath.Base(workflow.Path)
+		workflowURL := fmt.Sprintf("https://%s/%s/actions/workflows/%s",
+			hostname,
+			ghrepo.FullName(repo),
+			baseName)
+		if opts.IO.IsStdoutTTY() {
+			fmt.Fprintf(opts.IO.Out, "Opening %s in your browser.\n", utils.DisplayURL(workflowURL))
+		}
+		return utils.OpenInBrowser(workflowURL)
+	}
+
+	yaml, err := getWorkflowContent(client, repo, workflow)
+	if err != nil {
+		return fmt.Errorf("could not get workflow file content: %w", err)
+	}
+
+	theme := opts.IO.DetectTerminalTheme()
+	markdownStyle := markdown.GetStyle(theme)
+	if err := opts.IO.StartPager(); err != nil {
+		fmt.Fprintf(opts.IO.ErrOut, "starting pager failed: %v\n", err)
+	}
+	defer opts.IO.StopPager()
+
+	if !opts.Raw {
+		cs := opts.IO.ColorScheme()
+		out := opts.IO.Out
+
+		fmt.Fprintln(out, cs.Bold(workflow.Name))
+		fmt.Fprintf(out, "%s ID: %s", cs.Gray(workflow.Path), cs.Cyanf("%d", workflow.ID))
+
+		codeBlock := fmt.Sprintf("```yaml\n%s\n```", yaml)
+		rendered, err := markdown.RenderWithOpts(codeBlock, markdownStyle,
+			markdown.RenderOpts{
+				markdown.WithoutIndentation(),
+				markdown.WithoutWrap(),
+			})
+		if err != nil {
+			return err
+		}
+		_, err = fmt.Fprint(opts.IO.Out, rendered)
+		return err
+	}
+
+	if _, err := fmt.Fprint(opts.IO.Out, yaml); err != nil {
+		return err
+	}
+
+	if !strings.HasSuffix(yaml, "\n") {
+		_, err := fmt.Fprint(opts.IO.Out, "\n")
+		return err
+	}
+
+	return nil
+}
+
+func promptWorkflows(client *api.Client, repo ghrepo.Interface) (*shared.Workflow, error) {
+	workflows, err := shared.GetWorkflows(client, repo, 10)
```

Comment:
What was the reasoning behind limiting this to 10 workflows?

Reply:
totally arbitrary limit, i can bump it.